torch/_dynamo/variables/builder.py

```diff
@@ -645,17 +645,28 @@ def wrap_module(self, value: torch.nn.Module):
         elif getattr(value, "_is_fsdp_managed_module", False) or issubclass(
             value.__class__, torch.nn.parallel.distributed.DistributedDataParallel
         ):
-            if getattr(value, "_is_fsdp_managed_module", False):
-                # Note: we can't do this assert inside FSDP constructor,
-                # since we don't know yet whether dynamo will be used
-                assert getattr(
-                    value, "_fsdp_use_orig_params", False
-                ), "Dynamo only supports FSDP with use_orig_params=True"
+            return UnspecializedNNModuleVariable(
+                value, guards=self.make_guards(GuardBuilder.TYPE_MATCH)
+            )
+        elif getattr(value, "_is_fsdp_managed_module", False):
+            # Note: we can't do this assert inside FSDP constructor,
+            # since we don't know yet whether dynamo will be used
+            assert getattr(
+                value, "_fsdp_use_orig_params", False
+            ), "Dynamo only supports FSDP with use_orig_params=True"
+
+            guard_create_fns = [GuardBuilder.TYPE_MATCH]
+            if torch._dynamo.config.skip_fsdp_guards:
+                guard_create_fns = [GuardBuilder.ID_MATCH]
```

Comment:
Discussed offline - let's add a comment about the uni-directionality of modules going from no fsdp to fsdp, that's enough for now.